cub/test/test_device_run_length_encode.cu

```diff
@@ -1,964 +0,0 @@
-
-  CubDebugExit(dispatch_t::Dispatch(d_temp_storage,
-                                    temp_storage_bytes,
-                                    d_in_keys,
-                                    d_offsets,
-                                    d_lengths,
-                                    d_num_runs,
-                                    equality_op_t{},
-                                    elements,
-                                    0));
-
-  thrust::device_vector<std::uint8_t> temp_storage(temp_storage_bytes);
-  d_temp_storage = thrust::raw_pointer_cast(temp_storage.data());
-
-  CubDebugExit(dispatch_t::Dispatch(d_temp_storage,
-                                    temp_storage_bytes,
-                                    d_in_keys,
-                                    d_offsets,
-                                    d_lengths,
-                                    d_num_runs,
-                                    equality_op_t{},
-                                    elements,
-                                    0));
-  CubDebugExit(cudaDeviceSynchronize());
-  const int actual_num_runs = num_runs[0];
-  const int first_length    = lengths[0];
-  const int first_offset    = offsets[0];
-
-  AssertEquals(non_trivial_runs, actual_num_runs);
-  AssertEquals(first_length, magic_number);
-  AssertEquals(first_offset, magic_number);
-}
-
-//---------------------------------------------------------------------
-// Main
-//---------------------------------------------------------------------
-
-/**
- * Main
- */
-int main(int argc, char **argv)
-{
-  int num_items = -1;
-
-  // Initialize command line
-  CommandLineArgs args(argc, argv);
-  g_verbose = args.CheckCmdLineFlag("v");
-  args.GetCmdLineArgument("n", num_items);
-  args.GetCmdLineArgument("i", g_timing_iterations);
-
-  // Print usage
-  if (args.CheckCmdLineFlag("help"))
-  {
-    printf("%s "
-           "[--n=<input items> "
-           "[--i=<timing iterations> "
-           "[--device=<device-id>] "
-           "[--v] "
-           "\n",
-           argv[0]);
-    exit(0);
-  }
-
-  // Initialize device
-  CubDebugExit(args.DeviceInit());
-  printf("\n");
-
-  // %PARAM% TEST_CDP cdp 0:1
-
-  // Run tests with leading/trailing NaNs, where (NaN == NaN) is false
-  TestNaNs<float>();
-  TestNaNs<double>();
-
-  // Test different input types
-  TestSize<signed char, int, int>(num_items);
-  TestSize<short, int, int>(num_items);
-  TestSize<int, int, int>(num_items);
-  TestSize<long, int, int>(num_items);
-  TestSize<long long, int, int>(num_items);
-  TestSize<float, int, int>(num_items);
-  TestSize<double, int, int>(num_items);
-
-  TestSize<uchar2, int, int>(num_items);
-  TestSize<uint2, int, int>(num_items);
-  TestSize<uint3, int, int>(num_items);
-  TestSize<uint4, int, int>(num_items);
-  TestSize<ulonglong4, int, int>(num_items);
-  TestSize<TestFoo, int, int>(num_items);
-  TestSize<TestBar, int, int>(num_items);
-
-  TestTrivialRunsInFirstTile<false>();
```

Comment:
question: is there an alternative to the `TestTrivialRunsInFirstTile`?

Reply:
I am not sure what it is actually testing and whether that is not already handeled by another test

Reply:
The test in question is targeting a specific behavior that has low probability of appearing in random input.

Reply:
Can you give me a description what it tests, because I am not really sure

Reply:
Iirc, it's a test case where the first tile (i.e., the first `ITEMS_PER_THREAD*BLOCK_THREAD` items) contains only trivial runs. It's a case that was just recently fixed here: https://github.com/NVIDIA/cccl/pull/294

Reply:
I have ported the specific test case